system/Database/BasePreparedQuery.php

```diff
@@ -147,17 +147,18 @@ abstract public function _execute(array $data): bool;
     abstract public function _getResult();
 
     /**
-     * Explicitly closes the statement.
+     * Explicitly closes the prepared statement.
      */
-    public function close()
+    public function close(): bool
     {
-        if (! is_object($this->statement) || ! method_exists($this->statement, 'close')) {
-            return;
-        }
-
-        $this->statement->close();
+        return $this->_close();
```

Comment:
It seems to me PHPCPD didn't detect the amount of code duplication in the implementations of `_close()`. And the common pattern is (1) setting `$errors` to `true`, (2) checking existence of `$this->statement`, (3) doing the driver specific closing, (4) unsetting `$this->statement`, and (5) return the `$errors`.

Would it be better to do this?
```php
public function close(): bool
{
    if (! isset($this->statement)) {
        return true;
    }

    try {
        return $this->_close();
    } finally {
        unset($this->statement);
    }
}

abstract protected function _close(): bool;
```

Reply:
That's right. I would like to see these changes.

Reply:
OK, done.  I've moved duplication in the implementations of _close() to the base class as suggested